src/storage/remote/protocol.c

```diff
@@ -158,69 +106,214 @@ storageRemoteFeatureProtocol(const VariantList *paramList, ProtocolServer *serve
         }
 
         // Return storage features
-        protocolServerWriteLine(server, jsonFromStr(storagePathP(storage, NULL)));
-        protocolServerWriteLine(server, jsonFromUInt64(storageInterface(storage).feature));
+        PackWrite *result = protocolPackNew();
+        pckWriteStrP(result, storagePathP(storage, NULL));
+        pckWriteU64P(result, storageInterface(storage).feature);
 
-        protocolServerResponse(server, NULL);
+        protocolServerDataPut(server, result);
+        protocolServerDataEndPut(server);
     }
     MEM_CONTEXT_TEMP_END();
 
     FUNCTION_TEST_RETURN_VOID();
 }
 
 /**********************************************************************************************************************************/
+typedef struct StorageRemoteInfoProcotolWriteData
+{
+    MemContext *memContext;                                         // Mem context used to store values from last call
+    time_t timeModifiedLast;                                        // timeModified from last call
+    mode_t modeLast;                                                // mode from last call
+    uid_t userIdLast;                                               // userId from last call
+    gid_t groupIdLast;                                              // groupId from last call
+    String *user;                                                   // user from last call
+    String *group;                                                  // group from last call
+} StorageRemoteInfoProtocolWriteData;
+
+// Helper to write storage info into the protocol. This function is not called unless the info exists so no need to write exists or
+// check for level == storageInfoLevelExists.
+//
+// Fields that do not change from one call to the next are omitted to save bandwidth.
+static void
+storageRemoteInfoProtocolPut(
+    StorageRemoteInfoProtocolWriteData *const data, PackWrite *const write, const StorageInfo *const info)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM_P(VOID, data);
+        FUNCTION_TEST_PARAM(STORAGE_INFO, info);
+    FUNCTION_TEST_END();
+
+    ASSERT(data != NULL);
+    ASSERT(info != NULL);
+
+    // Write type and time
+    pckWriteU32P(write, info->type);
+    pckWriteTimeP(write, info->timeModified - data->timeModifiedLast);
+
+    // Write size for files
+    if (info->type == storageTypeFile)
+        pckWriteU64P(write, info->size);
+
+    // Write fields needed for detail level
+    if (info->level >= storageInfoLevelDetail)
+    {
+        // Write mode
+        pckWriteModeP(write, info->mode, .defaultValue = data->modeLast);
```

Comment:
Just so we can use the same function as `InfoList` uses. So, yeah, a lot of these conditionals are noops but it means less code. So all the data is transmitted since there was no previous info to delta from.